src/modules/developer-guide/pages/documentation/Conventions.1.adoc

```diff
@@ -0,0 +1,260 @@
+====
+
+=== Multi-line description blocks
+
+Use description blocks when your *lists* start *getting long*.
+They firstly present the *key point* of the block, followed by a more *detailed description* that is shifted to the right.
+
+[source,adoc]
+----
+// Make the key point outline the full description.
+// Use four spaces to indent the description.
+[Key point here]::
+    [Description here].
+----
+
+To examplify:
+[source,adoc]
+----
+== Corteza documentation structure
+
+DevOps guide::
+    The DevOps guide covers all of the DevOps related bits, such as deploying and maintaining.
+    ...
+
+End-user guide::
+    The end-user guide serves as a reference for end-users that need help with using the Corteza's user interface.
+    ...
+
+Integrator guide::
+    The integrator guide serves as a reference for integrators -- people that want to build things with Corteza.
+    ...
+
+----
+
+[IMPORTANT]
+====
+Try not to overuse this.
+If your descriptions *get long*, it may be better to restructure and divide the text into several sub-sections instead.
+====
+
+== Including code snippets
+
+=== Short one-liners
+
+Use *short one-liners* when you wish to show specific *things inline*, such as *CLI commands*, object *properties*, and *variables*.
+
+=== Longer code snippets
+
+When working on examples or longer code snippets, use the source code syntax; `[source,{language here}]`; blocked with four dashes `----`.
+Please, see the below example.
+
+```
+[source,adoc]
+----
+// Things go here
+----
+```
+
+== Admonition
+
+Use the *admonition block* syntax when wanting to create admonitions; where you want *extra attention* to be drawn to.
+
+.You can use the following:
+* NOTE
+* TIP
+* IMPORTANT
+* WARNING
+* CAUTION
+
+These are pretty much self-explanatory.
+Do not overuse.
+
+== Diagrams
+
+*Use the https://plantuml.com/[PlantUML]* framework for any diagrams (unless there is an explicit reason for not doing so).
+This allows us to *keep things consistent*, *preserve version history* and *easy to maintain*.
+
+[TIP]
+====
+Move *larger diagrams* into *separate files* to avoid clutter and unreadable source.
+====
+
+Provide a good caption to summarize the *entire* diagram. 
+You shouldn't use more than one line for a diagram caption.
+
+[plantuml,diagram-name-here,svg,role=diagram-name]
+----
+@startuml
+
+' Diagram definition goes here
+
+@enduml
+----
+
+[TIP]
+====
+Larger diagrams may be harder to understand. 
+Do your best to *group elements together*, and use a nice *comment* to indicate it.
+====
+
+Diagram captions::
+    A caption *must be enough* for the reader to get an idea about *what goes on* in a diagram.
+    Make sure that the caption gives the reader enough insight to have an idea *what they can expect* to learn *from the diagram*.
+
+[IMPORTANT]
+====
+Despite the 1 sentence per line rule, *free to inline your captions* - ???
+
+// Any updates on whether you're able to do it line by line?
+
```

Comment:
Our rule here is one sentence per line; so:

```
text 1 be here.
text 2 be here.
```

With these diagrams (and other captions as well; that `.text here` thing before lists and images), I was not able to figure out how to make it work over multiple lines (not sure if possible).
For these cases, this is allowed:

```
text 1 be here. text 2 be here
```